src/coreclr/hosts/unixcoreruncommon/coreruncommon.cpp

```diff
@@ -339,6 +339,17 @@ int ExecuteManagedAssembly(
     nativeDllSearchDirs.append(":");
     nativeDllSearchDirs.append(clrFilesAbsolutePath);
 
+    void* hostpolicyLib = nullptr;
+    char* mockHostpolicyPath = getenv("MOCK_HOSTPOLICY");
+    if (mockHostpolicyPath)
+    {
+        hostpolicyLib = dlopen(mockHostpolicyPath, RTLD_LAZY);
+        if (hostpolicyLib == nullptr)
+        {
+            fprintf(stderr, "Failed to load mock hostpolicy at path '%s'. Error: %s", mockHostpolicyPath, dlerror());
```

Comment:
Again, if a tests requests the behavior and it can't be satisfied then it needs to fail/crash/format drive... something to indicate the test can't verify the intended scenario.